src/crypto/algorithms/olm.ts

```diff
@@ -209,8 +209,21 @@ class OlmDecryption extends DecryptionAlgorithm {
         //
         // In practice, it's not clear that the js-sdk would behave that way, so this may be only a defence in depth.
 
-        await this.crypto.deviceList.downloadKeys([event.getSender()!], false);
-        const senderKeyUser = this.crypto.deviceList.getUserByIdentityKey(olmlib.OLM_ALGORITHM, deviceKey);
+        let senderKeyUser = this.crypto.deviceList.getUserByIdentityKey(olmlib.OLM_ALGORITHM, deviceKey);
+        if (senderKeyUser == undefined) {
+            // Await for any ongoing key query fetches for the user before trying the lookup again.
```

Comment:
```suggestion
            // Wait for any pending key query fetches for the user to complete before trying the lookup again.
```
because English is awful